### PR TITLE
Fix segfault if OSSL_PARAM_BLD_push_BN_pad() is called with NULL BN

### PR DESCRIPTION
**Pull request by t8m**
Related to #21935

This fixes the segfault on the master branch but there is still one remaining issue which is on all branches.

##### Checklist

- [x] tests are added or updated

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

File: crypto/param_build.c [hunk 1]
```diff
@@ -233,8 +233,8 @@ static int push_BN(OSSL_PARAM_BLD *bld, const char *key,
 int OSSL_PARAM_BLD_push_BN(OSSL_PARAM_BLD *bld, const char *key,
                            const BIGNUM *bn)
 {
-    if (BN_is_negative(bn))
-        return push_BN(bld, key, bn, bn == NULL ? 0 : BN_num_bytes(bn) + 1,
+    if (bn != NULL && BN_is_negative(bn))
```
Comment (nhorman) [inline crypto/param_build.c:236]: This check is sane, certianly, but shouldn't we already be protected from this by the NULL check in OPENSSL_sk_push?

Comment (mattcaswell) [reply crypto/param_build.c:236]: `BN_is_negative()` is not safe for use with  NULL pointer

Comment (nhorman) [reply crypto/param_build.c:236]: @mattcaswell yeah, that makes sense, my question is: should we allow the placing of NULL pointers on a stack_of type variant in the first place?  All the uses of sk_<type>_num seem to assume that all items on a stack are non-null.  Would it be better to add a check to OPENSSL_sk_push to return a failure if the data pointer being pushed/inserted is NULL?

Comment (t8m) [reply crypto/param_build.c:236]: This is a public API and it allows NULL BIGNUMs in 3.1/3.0 so this is necessary.

**Comment by t8m** [comment]
Merged to the master branch. Thank you for the reviews.

**Pull request closed without merging**